src/expr/src/relation/mod.rs

```diff
@@ -1152,6 +1152,14 @@ pub enum JoinImplementation {
     /// Each plan starts from the corresponding index, and then in sequence joins
     /// against collections identified by index and with the specified arrangement key.
     DeltaQuery(Vec<Vec<(usize, Vec<ScalarExpr>)>>),
+    /// Perform a semi join of a single input to a constant literal
+    ///
+    /// Technically, both differential and delta query implementations will
+    /// also perform semijoins if the semijoin would shrink the size of the
+    /// arranged inputs. A join is designated as having this implementation
+    /// in the single input case where neither `Differential` nor `DeltaQuery`
+    /// make sense.
+    Semijoin,
```

Comment:
My understanding of the `Differential` plan is that it specifies for each input which arrangement to use. Is there a reason that the `Semijoin` variant shouldn't just be the `Differential` variant with the appropriate arrangements?

Reply:
`Differential` specifies which arrangement to use for each input _except_ for the first one:
```
    /// The first argument indicates the index of the starting collection, and
    /// the sequence that follows lists other relation indexes, and the key for
    /// the arrangement we should use when joining it in.
    ///
    /// Each collection index should occur exactly once, either in the first
    /// position or somewhere in the list.
    Differential(usize, Vec<(usize, Vec<ScalarExpr>)>)
```

Reply:
That's true, but couldn't you use the constant collection as the first argument, and the specific arrangement you want to select from as the second argument?

Reply:
I had thought that you wanted the constants in the join constraints (see your Mar 4 comment above). Since the constants are in the join constraints, there is no constant collection, and often there is only one input.